documentation/kernel-semantics.tex

```diff
@@ -531,36 +531,16 @@ \subsection{Main Environment}
 We introduce the main environment to support static and library field extraction and assignment.
 This environment is visible during the execution of a \kernel{} program, therefore it should be in the left and right-hand side of all the transition presented in the paper.
 In order to simplify the transitions, we assume the main environment is implicit.
-Similar to Sectionn~\ref{subsec:env-definition}, the main environment can be extended and a field can be looked up in it.
-
-\subsubsection{Extend}
-\label{subsubsec:extend-main-env}
-
-The operator ``$\extendMain$'' creates a new main environment by extending the main environment with a new binding for the provided static field to a fresh location for the provided value.
-The newly created main environment overrides the current main environment.
-This operator has two implicit operands: the store and the main environment of the CESK machine.
-\begin{align*}
-    &\extendMain \in \dmenv \times \dfield \times \dval \rightarrow  \dmenv,\\
-    &\extendMain({\field},\,{\val}) = \mainenv',
-\end{align*}
-
-\noindent where
-\begin{align*}
-    & \field \in \dfield,\\
-    & \loc \text{ --- fresh location in the store},\\
-    & !\mainenv'(\field') =\begin{cases}
-        !\loc = \val, & \text{if }\field' = \field,\\
-        !\mainenv(\field'), & \text{otherwise}.
-    \end{cases}
-\end{align*}
-After the operation, the main environment is $\mainenv'$.
+The main environment is created before the execution of the main function and it is defined for all static and library fields.
+Similar to Section~\ref{subsec:env-definition}, static and library fields can be looked up in the main environment.
 
+For each static or library field $\field$, we define $\mainenv(\field) = \loc$, where $\loc$ is a fresh location in the store.
+In \kernel{} static and library fields are initialized when they are first accessed, hence on creation of $\mainenv$ the fresh locations will contain a special value, $\deref{\loc} = \emptyset, \loc \in \dlocation$.
```

Comment:
Indeed, I overlooked that part, however I might be misunderstanding what you suggest and I would appreciate a clarification.

I am not sure what you mean by "the global environment function to return the values from Empty + Location": my understanding is that the environment returns a location.

Are you suggesting to define the main environment so that when a static field is first accessed, the main env function returns Empty, and on subsequent calls it would return a location. I think this implies that the main environment will need to be mutable and we don't want that.

If you ment that we need to modify the "!" operator on the store to return either a Value or Empty, I agree and I indeed overlooked that part.

I will add/modify according to your response!